exporting/read_config.c

```diff
@@ -0,0 +1,354 @@
+{
+    _CONNECTOR_INSTANCE *local_ci;
+
+    if (!strcmp(section, CONFIG_SECTION_EXPORTING))
+        return appconfig_get_boolean(root, CONFIG_SECTION_EXPORTING, name, default_value);
+
+    local_ci = find_instance(section);
+
+    if (!local_ci)
+        return 0;       // TODO: Check if it is meaningful to return default_value
+
+    return appconfig_get_boolean(
+        root,
+        local_ci->instance_name,
+        name,
+        appconfig_get_boolean(
+            root,
+            local_ci->connector_name,
+            name,
+            appconfig_get_boolean(root, CONFIG_SECTION_EXPORTING, name, default_value)));
+}
+
+long long expconfig_get_number(struct config *root, const char *section, const char *name, long long default_value)
+{
+    _CONNECTOR_INSTANCE *local_ci;
+
+    if (!strcmp(section, CONFIG_SECTION_EXPORTING))
+        return appconfig_get_number(root, CONFIG_SECTION_EXPORTING, name, default_value);
+
+    local_ci = find_instance(section);
+
+    if (!local_ci)
+        return 0;   // TODO: Check if it is meaningful to return default_value
+
+    return appconfig_get_number(
+        root,
+        local_ci->instance_name,
+        name,
+        appconfig_get_number(
+            root,
+            local_ci->connector_name,
+            name,
+            appconfig_get_number(root, CONFIG_SECTION_EXPORTING, name, default_value)));
+}
+
+/*
+ * Get the next connector instance that we need to activate
+ *
+ * @param @target_ci will be filled with instance name and connector name
+ *
+ * @return  - 1 if more connectors to be fetched, 0 done
+ *
+ */
+
+int get_connector_instance(struct connector_instance *target_ci)
+{
+    static _CONNECTOR_INSTANCE *local_ci = NULL;
+    _CONNECTOR_INSTANCE *global_connector_instance;
+
+    global_connector_instance = find_instance(NULL);       // Fetch head of instances
+
+    if (unlikely(!global_connector_instance))
+        return 0;
+
+    if (target_ci == NULL) {
+        local_ci = NULL;
+        return 1;
+    }
+    if (local_ci == NULL)
+        local_ci = global_connector_instance;
+    else {
+        local_ci = local_ci->next;
+        if (local_ci == NULL)
+            return 0;
+    }
+
+    strcpy(target_ci->instance_name, local_ci->instance_name);
+    strcpy(target_ci->connector_name, local_ci->connector_name);
+
+    return 1;
+}
+
+/**
+ * Select Type
+ *
+ * Select the connector type based on the user input
+ *
+ * @param type is the string that defines the connector type
+ *
+ * @return It returns the connector id.
+ */
+BACKEND_TYPE exporting_select_type(const char *type)
+{
+    if (!strcmp(type, "graphite") || !strcmp(type, "graphite:plaintext")) {
+        return BACKEND_TYPE_GRAPHITE;
+    } else if (!strcmp(type, "opentsdb") || !strcmp(type, "opentsdb:telnet")) {
+        return BACKEND_TYPE_OPENTSDB_USING_TELNET;
+    } else if (!strcmp(type, "opentsdb:http") || !strcmp(type, "opentsdb:https")) {
```

Comment:
Hi @vlvkobal ,

I verified that you are keeping the option for `opentsd:https`, but I could not see any call for OpenSSL library in the new code, are you planning to do this late? O r do you prefer I bring this when it is merged?

Reply:
I'm going to do that later.

Reply:
All right, thanks!